packages/markdown-it-synapse/tsup.config.ts

```diff
@@ -0,0 +1,29 @@
+import { defineConfig } from 'tsup'
+
+const sharedConfig = {
+  entry: { index: './src/index.ts' },
+  minify: true,
+  dts: true,
+  clean: true,
+}
+
+const globalName = 'markdownitSynapse'
+
+export default defineConfig([
+  {
+    ...sharedConfig,
+    format: ['esm', 'cjs'],
+  },
+  {
+    ...sharedConfig,
+    format: ['iife'],
+    globalName: globalName,
+    esbuildOptions: (options, context) => {
+      // Add footer so the default export is the plugin function, which maintains behavior of previously released bundles
+      // See https://github.com/evanw/esbuild/issues/1182#issuecomment-1011414271
+      options.footer = {
+        js: `${globalName} = Object.assign(${globalName}.default, ${globalName})`,
+      }
```

Comment:
In short, esbuild won't let us make the IIFE global a function with properties. This is a breaking change, so we tack on this footer to merge the properties into the function object, and assign it to the global. This prevents the breaking change.